cola/widgets/selectcommits.py

```diff
@@ -145,10 +145,12 @@ def commit_oid_double_clicked(self, _item):
 
     def search_list(self, text):
         search_items = self.commits.findItems(text, Qt.MatchContains)
```

Comment:
I think it could become more like
```python
# Only pre-hide items unless we are going to show all items anyways
if !text.empty: # <-- I don't know python, just ruby, please adjust
    for i in range(self.commits.count()):
        self.commits.item(i).setHidden(True)

search_items = self.commits.findItems(text, Qt.MatchContains)
for items in search_items:
    items.setHidden(False)
```

Reply:
Otherwise we may end up with an incomplete list if you ping-pong between empty text and some text

Reply:
Makes sense. Made the change.